guide/src/trait_bounds.md

```diff
@@ -180,61 +179,55 @@ This wrapper will also perform the type conversions between Python and Rust.
 # impl Model for UserModel {
 #  fn set_variables(&mut self, var: &Vec<f64>) {
 #      println!("Rust calling Python to set the variables");
-#      let gil = Python::acquire_gil();
-#      let py = gil.python();
-#      let values: Vec<f64> = var.clone();
-#      let list: PyObject = values.into_py(py);
-#      let py_model = self.model.as_ref(py);
-#      py_model
-#          .call_method("set_variables", (list,), None)
-#          .unwrap();
+#      Python::with_gil(|py| {
+#          let values: Vec<f64> = var.clone();
+#          let list: PyObject = values.into_py(py);
+#          let py_model = self.model.as_ref(py);
+#          py_model
+#              .call_method("set_variables", (list,), None)
+#              .unwrap();
+#      })
 #  }
 #
 #  fn get_results(&self) -> Vec<f64> {
 #      println!("Rust calling Python to get the results");
-#      let gil = Python::acquire_gil();
-#      let py = gil.python();
-#      self
-#          .model
-#          .as_ref(py)
-#          .call_method("get_results", (), None)
-#          .unwrap()
-#          .extract()
-#          .unwrap()
+#      Python::with_gil(|py| {
+#          self.model
+#              .as_ref(py)
+#              .call_method("get_results", (), None)
+#              .unwrap()
+#              .extract()
+#              .unwrap()
+#      })
 #  }
 #
 #  fn compute(&mut self) {
 #      println!("Rust calling Python to perform the computation");
-#      let gil = Python::acquire_gil();
-#      let py = gil.python();
-#      self.model
-#          .as_ref(py)
-#          .call_method("compute", (), None)
-#          .unwrap();
+#      Python::with_gil(|py| {
+#          self.model
+#              .as_ref(py)
+#              .call_method("compute", (), None)
+#              .unwrap();
+#      })
+#
 #  }
 # }
 
 #[pymethods]
 impl UserModel {
-    pub fn set_variables(&mut self, var: Vec<f64>) -> PyResult<()> {
+    pub fn set_variables(&mut self, var: Vec<f64>) {
```

Comment:
This method should take `py: Python`.

Reply:
I agree, but it's actually quite hard to solve.

The trait signature `impl Model for UserModel` has no way to pass a `Python` into the trait methods. The idea for this example is to show how to wrap a Rust trait which doesn't know anything about `Python`, so I don't think changing the trait definition is a solution either.

For now I would like to leave it as it is, but if we come up with a good idea how to clean this up later I would be happy to resolve it. (I'll create an issue so we remember this.)